modules/algebra/src/main/java/org/eclipse/imagen/media/algebra/AlgebraOpImage.java

```diff
@@ -300,7 +300,12 @@ public AlgebraOpImage(
         caseC = hasNoData && !hasROI;
 
         // Set flag to permit in-place operation.
-        permitInPlaceOperation();
+        if (Operator.XOR != op) {
+            // TODO: The XOR operator is having troubles with the permit in-place
+            //  operation since it uses the first source as destination and computing
+            //  Xor on itself
+            permitInPlaceOperation();
```

Comment:
Interesting, what were the old add/subtract and friends doing? In place or new image?

Reply:
Looks like they were in place too.